controllers/buildpipeline/buildpipeline_adapter.go

```diff
@@ -63,19 +63,41 @@ func NewAdapter(pipelineRun *tektonv1.PipelineRun, component *applicationapiv1al
 
 // EnsureSnapshotExists is an operation that will ensure that a pipeline Snapshot associated
 // to the build PipelineRun being processed exists. Otherwise, it will create a new pipeline Snapshot.
-func (a *Adapter) EnsureSnapshotExists() (controller.OperationResult, error) {
-	var err error
+func (a *Adapter) EnsureSnapshotExists() (result controller.OperationResult, err error) {
+	// a marker if we should remove finalizer from build PLR
+	var canRemoveFinalizer bool
+
 	defer func() {
 		err = tekton.AnnotateBuildPipelineRunWithCreateSnapshotAnnotation(a.context, a.pipelineRun, a.client, err)
 		if err != nil {
 			a.logger.Error(err, "Could not add create snapshot annotation to build pipelineRun", h.CreateSnapshotAnnotationName, a.pipelineRun)
+			// requeue when err is not nil
+			result = controller.OperationResult{
+				RequeueDelay:   0,
+				RequeueRequest: true,
+				CancelRequest:  false,
+			}
```

Comment:
Logic looks good but I'm curios if we could use this to make readability better?
```suggestion
			result, err = controller.RequeueWithError(err)
```

Reply:
Updated. Thanks.